ONSdigital/ras-frontstage
# RAS-739 Add modal to base template

# What and why?
This PR adds the timeout modal.
# How to test?
It's best to test this locally, for 2 reasons, csrf is turned off in dev and the timeout is set to 60 minutes, turning on csrf in dev isn't an issue, but the timeout is hardcoded which is a pain.

Now that you have it running locally, go into the code and change this to something sensible https://github.com/ONSdigital/ras-frontstage/blob/main/frontstage/common/session.py#L87 like 65, this means the modal will pop up after 5 seconds making it a lot easier to test. Basically you now what to interact with the app and the modal, recommended tests

1. Navigated around the app clicking with-inside the 5 seconds(if you set it to 65), make sure the timeout doesn't appear
2. Let the modal pop up then press the continue and make sure the session is extended (http://localhost:8082/session/expires-at is useful), repeat however many times you like
3. Let the modal run down and check it logs you out
4. Check the modal works without focus, i.e log in to frontstage then go to another page and make sure it is logged out after you come back after 70 seconds
5. Anything else you can think off


# Jira
